numpyro/distributions/distribution.py

```diff
@@ -150,9 +152,9 @@ def __init__(self, batch_shape=(), event_shape=(), validate_args=None):
                     continue
                 if is_dependent(constraint):
                     continue  # skip constraints that cannot be checked
-                is_valid = jnp.all(constraint(getattr(self, param)))
+                is_valid = constraint(getattr(self, param))
                 if not_jax_tracer(is_valid):
-                    if not is_valid:
+                    if not np.all(is_valid):
```

Comment:
For my understanding, if we were to always cast a python bool to a numpy or jax bool type (i.e. the default mask would be a `jnp.array(True)`), we would not have needed any change here. Is that correct?

Reply:
We we need this change because under `jit`. `jnp.all(True)`, `device_put(True)` will create a tracer. So validation code won't activate.
```python
from jax import device_put, jit
import jax.numpy as jnp
from numpyro.util import not_jax_tracer

def f():
    assert not not_jax_tracer(jnp.all(True))
    assert not not_jax_tracer(device_put(True))

jit(f)()  # pass
```

You are right that if we cast `mask` in `_validate_sample` method to a device array, then the error will be fixed. In that case, no validation for parameters/sample are activated under `jit`. It is fine for single chain mcmc because we trace the model outside of `jit` context, so if there are some wrong specification, the users will get the warning/error. But for multi chain mcmc, no validation will be activated because we run everything under `pmap`.

Reply:
Thanks for explaining about `pmap`, I missed that. In that case, is it correct to say that for distributions that have all parameters as jax devices (no python scalars), there won't be any validation performed under `pmap` because `not_jax_tracer(param)` will be true?

Reply:
Yes, that's true.